airflow/providers/common/sql/operators/sql.py

```diff
@@ -585,41 +622,70 @@ def __init__(
         partition_clause: str | None = None,
         conn_id: str | None = None,
         database: str | None = None,
+        accept_none: bool = True,
```

Comment:
before these changes, when a query was returning None the task was failing, by setting  `accept_none` default value to true I think will change the old behaviour i.e now a task may not fail. do we need to worry about backward compatibility and maybe set this false by default, wdyt?

Reply:
@mag3141592 requested this and may be able to speak better about it, but basically the old behavior is wrong from a correctness standpoint.

Reply:
`SQLTableCheckOperator` didn't have an `accept_none` param previously, but `SQLColumnCheckOperator` did and its default is `True`. Ideally these are consistent, but yes that poses an issue with backward compatibility.

So I'm not sure what the priority is, consistency or backward compatibility? I would lean toward consistency, but I also made the request 🤷‍♀️

Reply:
We can treat it as a bug fix. It's really unlikely someone's workflow depends on None being returned here. You'd rather make everything to get a number as result of the check. @eladkal - WDYT?

Reply:
Why can't we deprecate first?
SQLTableCheckOperator - We can add new parameter for with default value False
SQLColumnCheckOperator - Raise deprecation warning that default value will be changed in future versions to False.

In the future we can release common.sql provider with major release that changes the default value.
We are very careful with major releases of common.sql but this one will be fine. WDYT @potiuk ?

Reply:
Sure we can do it this way, but I find it a bit odd to add new parameter and deprecate it immediately. Somehow like - we are adding this new feature there, but BTW it's deprecated. 

If we deprecate it in easy way, there will be no way to distinguish if user deliberately passed False, or it was the default one. 

False is immutable so you cannot use the technique from https://stackoverflow.com/questions/11251119/is-there-any-way-to-know-if-the-value-of-an-argument-is-the-default-vs-user-spe where you could compare identity of passed object. As far as I know. If you make `accept_none=False` the default. for all practical purposes (other than introspecting the source code from the call stack) - `SQLTableCheckOperator(...., accept_none=False)` is indistinguishable from `SQLTableCheckOperator(...., ). Which means that if we deprecate accept_none=False, there will be no way to pass False without a Deprecation warnings - i.e. user who would like to use False will have no way to remove the warning. 

You could potentially pass a sentinel object and treat it as False (and deprecate), but that's really makes things like that complex and ugly (also confusing user):

```
SQLTableCheckOperator(, accept_none: bool | Literal["TREAT_ME_AS_FALSE"] = "TREAT_ME_AS_FALSE") 
```


But if you think it's likely that someone would fall into backwards compatibility there, and it's worth it - yeah. it could be done.

Reply:
Yeah. I am usually in the camp of "breaking change is only determined by how likely it is to change other's workflow" :).  I **think** this case is where it's not very likely somoene treats none as false (though I must say I have never used that operator). 

@pankajastro @denimalpaca - > maybe you have some examples of how people are using the operator and elaborate on how likely it is that the breaking behaviour will kick in ?

Reply:
sorry, I don't have example I was just worried that this may surprise some users.
@mag3141592 do you have some examples of how we are using this?

Reply:
An example of how we're using them, is by testing our incremental table change before loading them into our prod table. If there is no incremental table `SQLTableCheckOperator` currently fails and `SQLColumnCheckOperator` currently passes (because the `accept_none = True` by default). If the defaults are False, then we'd just need to update our end to set `accept_none = True`. 

Sometimes an empty table is indicative of an error and sometimes it isn't, so needed a param for communicating that.